lib/launchers/ocm.rb

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env ruby
+
+lib_path = File.expand_path(File.dirname(File.dirname(__FILE__)))
+unless $LOAD_PATH.any? {|p| File.expand_path(p) == lib_path}
+  $LOAD_PATH.unshift(lib_path)
+end
+
+require 'common'
+require 'json'
+require 'tmpdir'
+
+module BushSlicer
+  class OCM
+    include Common::Helper
+
+    attr_reader :config
+    attr_reader :token, :token_file, :url, :region, :version, :nodes, :lifespan, :cloud, :cloud_opts, :multi_az
+
+    def initialize(**options) 
+      service_name = ENV['OCM_SERVICE_NAME'] || options[:service_name] || 'ocm'
+      @opts = default_opts(service_name)&.merge options
+      unless @opts
+        @opts = options
+      end
+
+      # OCM token is mandatory
+      # it can be defined by token or by token_file
+      @token = ENV['OCM_TOKEN'] || @opts[:token]
+      @token_file = @opts[:token_file]
+      unless @token
+        if @token_file
+          token_file_path = expand_private_path(@token_file)
+          @token = File.read(token_file_path)
+        else
+          raise "You need to specify OCM token by 'token' or by 'token_file'"
+        end
+      end
+
+      # region is mandatory
+      # in the future we can extend support for other clouds, e.g. GCP and ARO
+      @region = ENV['OCM_REGION'] || ENV['AWS_REGION'] || @opts[:region]
+
+      # url defines the OCM environment (prod, integration or stage)
+      # currently, the url is ignored as many teams use the stage environment
+      @url = ENV['OCM_URL'] || @opts[:url] || 'https://api.stage.openshift.com'
+
+      # openshift version is optional
+      @version = ENV['OCM_VERSION'] || @opts[:version]
+
+      # number of worker nodes
+      # minimum is 2
+      # default value is 4
+      @nodes = ENV['OCM_NODES'] || @opts[:nodes]
```

Comment:
Seems like these variables select what cluster will be installed. I think variable names should be more like `OCM_INSTALL_NODES` or something like this.

My bigger concern though is that and OCM object is bound to an installed OCP cluster. I would expect to see an `OpenshiftClusterManager` class that contains interface to interact with OCM. And then an `OCMInstalledCluster` class which will represent an OCP cluster that has been installed via OCM.

In that way you will have separation of concerns about interfacing with OCM and operating on the cluster installed by it. e.g. first `OCMInstalledCluster` object will be created with the cluster definition where `ocm_installed_cluster.version` will return version you want,  `ocm_installed_cluster.lifespan` will return configured lifepan, etc. And the `OpenshiftClusterManager` class will be able to install, remove and manage clusters based on `OCMInstalledCluster` objects.

Makes sense or am I missing something?

Reply:
It makes sense and I'm ok to refactor current implementation in the way you have described.
I did not notice anything similar for `amz`, `azure`, `openstack`, ... that's why I did it in such a straightforward way.
But before refactoring it, I'd like to be sure that having this among launchers and having `cloud_type=ocm` is a good approach.

Reply:
The other cloud classes deal with VMs, not clusters and one cloud management object creates potentially many VMs. In some classes we have separate VM classes, in some refer to them only with string names. You can keep OCP class in same file as the OCM class. 

I'm still not sure whether OCM should be among the IaaS providers. It is a quet different beast. On the other hand `cloud_type=ocm` sounds reasonable. Ideally I'd like to move IaaS providers to `/lib/iaas` but always keep on the back burner. 

I think you can keep file where it is now and it can be moved later if needed.

Reply:
Would you include a sample configuration into `config/config.yaml`?

Reply:
```
  # AWS account dedicated to OSD testing
  aws-osd:
    awscred: config/.awscred-osd
    host_opts:
      user: root
      class: SSHAccessibleHost
      ssh_private_key: config/keys/integration-qe.pem
    config_opts:
      region: us-east-2
    cloud_type: aws
  # Openshift Cluster Management for Openshift Dedicated (OSD)
  ocm:
    token_file: config/.ocm-token
    url: https://api.stage.openshift.com
    cloud: aws-osd
    region: us-east-2
    flavor: m5.xlarge
    nodes: 4
    lifespan: 8h
    cloud_type: ocm
 ```

Reply:
@akostadinov I was thinking of refactoring this OCM class as you have described and now I think that it is not needed. I do not implement anything like `OpenshiftClusterManager` as this is implemented by `ocm-cli` tool which I execute.
The class `ocm.rb` just takes a cluster configuration, transforms it to a json and posts this json via the `ocm-cli` tool.

Reply:
If this represents a asingle OCM cluster, then please rename class to `OCMCluster` and do not add it as an IaaS in `cloud_helper`. Given it relies on non-public script and probably moved to internal repo (e.g. the template repo), then it will not make sense anyway.

Reply:
Done